Issue 5585 - lib389 password policy DN handling is incorrect

droideck opened the pull request:
Description: After a migration between major DS versions, it can happen that already existing password policies will have 'cn' that contains a valid DN in double quote "". We need to strip the quotes before processing the DN with python-ldap.

Fixes: https://github.com/389ds/389-ds-base/issues/5585

Reviewed by: ?

tbordaz commented on dirsrvtests/tests/suites/password/password_policy_test.py:
```diff
@@ -1378,6 +1438,74 @@ def test_additional_corner_cases(topo, _policy_setup, _fixture_for_additional_ca
+    # And now, try to get the unusual subtree pwpolicy
+    people = pwp.get_pwpolicy_entry(f'ou=people,{DEFAULT_SUFFIX}')
+    people.replace('passwordhistory', 'off')
+    assert people.get_attr_val_utf8('passwordhistory') == 'off'
```
Would it be possible to delete users/pw_policy in teardown ?

droideck replied:
Done, I've separated part of the function into the fixture and have added a subtree removal as a tearDown there.

tbordaz reviewed (approved):
Fix looks good to me (just a minor comment)

mreynolds389 reviewed (approved):
Ack

droideck committed:
Issue 5585 - lib389 password policy DN handling is incorrect

Description: After a migration between major DS versions, it can happen
that already existing password policies will have 'cn' that contains
a valid DN in double quote "". We need to strip the quotes before
processing the DN with python-ldap.

Fixes: https://github.com/389ds/389-ds-base/issues/5585

Reviewed by: @tbordaz, @mreynolds389 (Thanks!)
